--- a/loader/src/external/plti/Makefile	2026-06-24 07:42:56.330325645 +0000
+++ b/loader/src/external/plti/Makefile	2026-06-24 07:49:02.836311463 +0000
@@ -2,6 +2,8 @@
 API_LEVEL ?= 26
 ARCH ?= arm64-v8a
 
+NDK_CFLAGS ?= 
+
 TOOLCHAIN = $(NDK_PATH)/toolchains/llvm/prebuilt/linux-x86_64
 SYSROOT = $(TOOLCHAIN)/sysroot
 
@@ -23,12 +25,12 @@
 CFLAGS_BASE = -std=c99 -DANDROID -fPIC -Isrc -Wpedantic -Wall -Wextra -Werror -Wformat -Wuninitialized -Wshadow -Wno-variadic-macros
 
 CFLAGS_debug = $(CFLAGS_BASE) -g -O0 -DPLTI_LOGGING
-CFLAGS_release = $(CFLAGS_BASE) -O3 -flto
+CFLAGS_release = $(CFLAGS_BASE) -O3 -flto -funsafe-math-optimizations
 
 BUILD_TYPE ?= debug
 out ?= build/$(BUILD_TYPE)/$(ARCH)
 
-CFLAGS = $(CFLAGS_$(BUILD_TYPE))
+CFLAGS = $(NDK_CFLAGS) $(CFLAGS_$(BUILD_TYPE)) -Wno-unused-command-line-argument
 
 SRCS = src/plti.c src/elf_util.c src/fd_utils.c
 OBJS = $(patsubst src/%.c,$(out)/%.o,$(SRCS))
